src/components/modals/modals.js

```diff
@@ -53,6 +68,19 @@ var dh2oModal = function (modalId, autoShow, callback = null) {
     }, 300)
   }
 
+  modal.isLocked = false
+
+  // call this to prevent the user from closing the modal. Note that hide()
+  // won't work when the modal is locked.
+  modal.lock = function () {
+    modal.isLocked = true
+    modal.querySelector('.modal-close-x').innerHTML = ''
+  }
+  modal.unlock = function () {
+    modal.isLocked = false
+    modal.querySelector('.modal-close-x').innerHTML = 'X'
```

Comment:
I thought we decided no X close buttons. Might need to remove completely the html as well.